src/libsystemd/sd-bus/bus-objects.c

```diff
@@ -1610,6 +1611,102 @@ static int vtable_member_compare_func(const struct vtable_member *x, const struc
+} names_flags;
+
+static bool names_are_valid(const char *signature, const char **names, names_flags *flags) {
+        int r;
+
+        if ((*flags & NAMES_FIRSTPART || *flags & NAMES_SINGLEPART) && **names != '\0')
+                *flags |= NAMES_PRESENT;
+
+        for (;*flags & NAMES_PRESENT;) {
+                size_t l;
+
+                if (!*signature)
+                        break;
+
+                r = signature_element_length(signature, &l);
+                if (r < 0)
+                        return false;
+
+                if (**names != '\0') {
+                        if (!member_name_is_valid(*names))
+                                return false;
+                        *names += strlen(*names) + 1;
+                } else if (*flags & NAMES_PRESENT)
+                        return false;
+
+                signature += l;
+        }
+        /* let's check if there are more argument names specified than the signature allows */
+        if (*flags & NAMES_PRESENT && **names != '\0' && !(*flags & NAMES_FIRSTPART))
+                return false;
+        *flags &= ~NAMES_FIRSTPART;
+        return true;
+}
+
+/* the current version of this struct is defined in sd-bus-vtable.h, but we need to list here the historical versions
+   to make sure the calling code is compatible with one of these */
+struct sd_bus_vtable_v241 {
+        uint8_t type:8;
+        uint64_t flags:56;
+        union {
+                struct {
+                        size_t element_size;
+                } start;
+                struct {
+                        const char *member;
+                        const char *signature;
+                        const char *result;
+                        sd_bus_message_handler_t handler;
+                        size_t offset;
+                } method;
+                struct {
+                        const char *member;
+                        const char *signature;
+                } signal;
+                struct {
+                        const char *member;
+                        const char *signature;
+                        sd_bus_property_get_t get;
+                        sd_bus_property_set_t set;
+                        size_t offset;
+                } property;
+        } x;
+};
+/* Structure size up to v241 */
+#define VTABLE_ELEMENT_SIZE_241 sizeof(struct sd_bus_vtable_v241)
+/* Current structure size */
+#define VTABLE_ELEMENT_SIZE sizeof(struct sd_bus_vtable)
+
+static int vtable_version(const sd_bus_vtable *vtable) {
+        if (vtable[0].x.start.element_size == VTABLE_ELEMENT_SIZE_241)
+                return 241;
+        return 0; /* current */
+}
+
+bool vtable_has_names(const sd_bus_vtable *vtable) {
+        if (vtable_version(vtable) == 241)
+                return false;
+        return true;
+}
+
+const sd_bus_vtable* next_vtable(const sd_bus_vtable *vtable, const sd_bus_vtable *v) {
```

Comment:
hmm, for symbols exported outside of this .c file, let's prefix things with "bus_", please (yeah, we don't follow this rule strictly current, but let's still try to) also, usually we place the subject first when naming, i.e. bus_vtable_next() sounds like a good name